File: src/frontend/bank/src/app/dtos/user-account-transaction-out.ts
// todo: interface?
// todo transaction creation or transaction out?
export class UserAccountTransactionOut {
  fromUserAccountId: string = '';
  toUserAccountId: string = '';
  amount: number = 0;

  public constructor(init?: Partial<UserAccountTransactionOut>) {
    Object.assign(this, init);
  }
}

File: src/main/java/ru/shibaev/bank/model/models/UserAccountTransactionOut.java
package ru.shibaev.bank.model.models;

import java.math.BigDecimal;
import java.util.UUID;

public class UserAccountTransactionOut {

    public UUID fromUserAccountId;
    public UUID toUserAccountId;
    public BigDecimal amount;
    // todo currency

    public UserAccountTransactionOut(UUID fromUserAccountId, UUID toUserAccountId, BigDecimal amount) {
        this.fromUserAccountId = fromUserAccountId;
        this.toUserAccountId = toUserAccountId;
        this.amount = amount;
    }
}
